contentstack/app-sdk
# feat: add default modal configuration

## Tasks

- [x] ~~Set `window.iframeRef` to the `window.document.body`~~
- [x] Set `window.postRobot` to `postRobot`
- [x] Add Modal module with `.setBackgroundElement()` method

## QA

Unit test cases have been added to test the following cases
- Extension object
  - should have a modal property
  - should add body to iframeRef if the user has not provided one
  - should not add body to iframeRef if the user has provided one
- Modal object
  - should add the first child of the body to iframeRef if the user has not provided one
  - should not add anything to iframeRef if it already exists.
  - should set iframeRef to the element provided by the user.
  - should have a modal property

fixes: https://contentstack.atlassian.net/browse/CS-26933
